api/config/v1alpha1/envoygateway_types.go

```diff
@@ -135,6 +141,58 @@ type RateLimitRedisSettings struct {
 	URL string `json:"url"`
 }
 
+// Extension defines the configuration for registering an extension to
+// the Envoy Gateway control plane.
+type Extension struct {
+	// Resources defines the set of K8s resources the extension will handle.
+	//
+	// +optional
+	Resources []GroupVersionKind `json:"resources,omitempty"`
+
+	// Hooks defines the set of hooks the extension supports
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Hooks []ExtensionHook `json:"hooks,omitempty"`
+
+	// Service defines the configuration of the extension service that the Envoy
+	// Gateway Control Plane will call through extension hooks.
+	//
+	// +kubebuilder:validation:Required
+	Service *ExtensionService `json:"service,omitempty"`
```

Comment:
should `service` be defined per hook or maybe this way makes it less repetitive , and in the future extension can be made plural

Reply:
we only allow one extension to be registered so don't see a real benefit in that